components/viaduct/src/backend/ffi.rs

```diff
@@ -144,22 +144,32 @@ mod callback_holder {
     /// Note: We only assign to this once.
     static CALLBACK_PTR: AtomicUsize = AtomicUsize::new(0);
 
+    ffi_support::static_assert!(
+        STATIC_ASSERT_USIZE_EQ_FUNC_SIZE,
+        std::mem::size_of::<usize>() == std::mem::size_of::<FetchCallback>()
+    );
+
+    ffi_support::static_assert!(
+        STATIC_ASSERT_USIZE_EQ_OPT_FUNC_SIZE,
+        std::mem::size_of::<usize>() == std::mem::size_of::<Option<FetchCallback>>()
+    );
+
     /// Get the function pointer to the FetchCallback. Panics if the callback
     /// has not yet been initialized.
     pub(super) fn get_callback() -> Option<FetchCallback> {
-        let ptr = CALLBACK_PTR.load(Ordering::SeqCst) as *const FetchCallback;
-        if ptr.is_null() {
-            None
-        } else {
-            Some(unsafe { *ptr })
-        }
+        let ptr_value = CALLBACK_PTR.load(Ordering::SeqCst);
+        unsafe { std::mem::transmute::<usize, Option<FetchCallback>>(ptr_value) }
     }
 
     /// Set the function pointer to the FetchCallback. Returns false if we did nothing because the callback had already been initialized
-    pub(super) unsafe fn set_callback(h: *const FetchCallback) -> bool {
-        let old_ptr = CALLBACK_PTR.compare_and_swap(0, h as usize, Ordering::SeqCst);
```

Comment:
Did the `as usize` cast fail on compile or during runtime? I'm not quite sure what the issue was about.

Reply:
Hard crash when it calls. Not exactly sure why but something something ARM calling convention.

Reply:
Or rather, I'm not exactly sure why it works on x86. This is the correct typing for this.